File: Perl/parse_fastqc.pl
#!/bin/perl

use Getopt::Long;

use strict;
use File::stat;
use Time::localtime;
use QCAnalysis::FastQC;
use QCAnalysis;
use QCAnalysis::DB;
use QCAnalysis::RunTable;



#use warning;

my %arguments = ();

my $opt_ret = GetOptions( \%arguments, 'input=s', 'db_config=s' );

print "Input: " . $arguments{input} . "\n";
#print "Sample ID: " . $arguments{sample} . "\n";
print "DB configuration: ". $arguments{db_config}."\n";
my $input = $arguments{input};

my $line;
my $module;
my $status;
my $config =  $arguments{db_config};

#my $analysis = QCAnalysis->new();
my $db = QCAnalysis::DB->new();

QCAnalysis::RunTable->add_header_scope("barcode", "analysis");

my @analysis =  QCAnalysis::RunTable->parse_file($input); 

#my $fast_qc_file = $analysis->get_property("path_to_fastqc");
#QCAnalysis::FastQC->parse_file($fast_qc_file, $analysis);

$db->connect($config);
foreach(@analysis){
 #  print "Anaysis: ".$_;
	my $fast_qc_file = $_->get_property("path_to_fastqc");
	if (-e $fast_qc_file) {
		QCAnalysis::FastQC->parse_file($fast_qc_file, $_);
		$db->insert_analysis($_);
	}else{
		print "WARN: Unable to read file $fast_qc_file\n";
	}
}
$db->disconnect();

File: python/parse_fastqc.py

import os, sys
import MySQLdb
import optparse
import QCAnalysis, FastQC, RunTable, DB

parser=optparse.OptionParser()
parser.add_option("-i", "--input", action="store", type="string", dest="input")
parser.add_option("-d", "--db_config", action="store", type="string", dest="db_config")

(options, args) = parser.parse_args()   #gets the arguments and options
options=options.__dict__
print "Input: " + options['input'] + "\n"
print "DB configuration: " + options['db_config'] + "\n"
input_metafile=options['input']
line=""; module=""; status=""; config=options['db_config']
print "Input : ", input_metafile
print "config : ", config


print "Creating database connection "


try:
	#read config file and get connection information
	dbhost=""; dbuser="";password=""; dbname=""
	fh=open(config)
	
	firstline=fh.readline().strip().split("\t")[1].split(";")
	dbhost=firstline[1].split("=")[1]
	dbname=firstline[0].split(":")[2]
	dbuser=fh.readline().strip().split("\t")[1]
	password=fh.readline().strip().split("\t")[1]
	#print dbhost, dbuser, password, dbname
	dbase_con= MySQLdb.connect(dbhost, dbuser, password, dbname)	#this is the object holding connection
	#dbase_con = MySQLdb.connect(db='statsdb')
	
except:
	print "Could not get connection. Please check the database name."
	exit(1)
RunTable.add_header_scope("barcode", "analysis")
list_of_objs=RunTable.parse_file(input_metafile)

#db.connect(config)
db=DB.dbase(dbase_con)



for obj in list_of_objs:
	print obj.data["property"]
	fast_qc_file = obj.get_property("path_to_analysis")
	print "fast_qc_file: ", fast_qc_file
	if os.path.exists(fast_qc_file):
		FastQC.parse_file(fast_qc_file, obj)
		print "Inserting the record now: "
		db.insert_analysis(obj)
	else:
		print "WARN: Unable to read file ", fast_qc_file

dbase_con.close()
